specs/opcode/3cEXTCODECOPY.md

```diff
@@ -0,0 +1,42 @@
+# EXTCODECOPY opcode
+
+## Procedure
+
+The `EXTCODECOPY` opcode pops `address`, `memory_offset`, `code_offset` and `size` from the stack. It then copies `size` bytes of code of `address` from an offset `code_offset` to the memory at the address `memory_offset`. For out-of-bound scenarios where `size > len(code) - code_offset`, EVM pads 0 to the end of the copied bytes.
+
+The gas cost of `EXTCODECOPY` opcode is dynamic and is based on the cost of memory expansion and copying (variable depending on the `size` copied to memory) and accessing the address (variable depending on whether the `address` is warm or not)
+
+## Circuit Behaviour
+
+The `EXTCODECOPY` circuit constrains the values popped from stack, call context/address read lookups and CopyTable lookups to verify the copy of bytes. The copy of a dynamic number of bytes is verified by the CopyCircuit outside the `EXTCODECOPY` gadget.
+
+The gadget then transits to the internal state of `CopyCodeToMemory`.
```

Comment:
This line is no longer needed

Reply:
Oops, I should have been careful. 
I Fixed in [03be899](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/257/commits/03be8994d3e706bde93bd2c9d0d276c98a93e182) :)